generator/src/test/java/com/scottlogic/deg/generator/generation/field_value_sources/CombiningFieldValueSourceEqualityTests.java

```diff
@@ -0,0 +1,84 @@
+package com.scottlogic.deg.generator.generation.field_value_sources;
+
+import com.scottlogic.deg.generator.utils.RandomNumberGenerator;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.IsNot.not;
+
+class CombiningFieldValueSourceEqualityTests {
+    @Test
+    public void shouldBeEqualIfUnderlyingSourcesAreTheSame(){
+        CombiningFieldValueSource a = new CombiningFieldValueSource(Arrays.asList(source(1), source(2)));
+        CombiningFieldValueSource b = new CombiningFieldValueSource(Arrays.asList(source(1), source(2)));
+
+        Assert.assertThat(a, equalTo(b));
+        Assert.assertThat(a.hashCode(), equalTo(b.hashCode()));
+    }
+
+    @Test
+    public void shouldBeUnequalIfUnderlyingSourcesAreTheSameButDifferentOrder(){
+        CombiningFieldValueSource a = new CombiningFieldValueSource(Arrays.asList(source(1), source(2)));
+        CombiningFieldValueSource b = new CombiningFieldValueSource(Arrays.asList(source(2), source(1)));
+
+        Assert.assertThat(a, not(equalTo(b)));
+    }
+
+    @Test
+    public void shouldBeUnequalIfUnderlyingSourcesAreDifferent(){
+        CombiningFieldValueSource a = new CombiningFieldValueSource(Arrays.asList(source(1), source(2)));
+        CombiningFieldValueSource b = new CombiningFieldValueSource(Arrays.asList(source(2), source(3)));
+
+        Assert.assertThat(a, not(equalTo(b)));
+    }
+
+    private FieldValueSource source(int hashCode){
+        return new MockFieldValueSource(hashCode);
+    }
+
+    private class MockFieldValueSource implements FieldValueSource{
```

Comment:
Is there not a way to use a mocking framework like Mockito to do this instead of manually?

Reply:
Unfortunately not, mockito doesn't support asserting/verifying `equals` or `hashCode`
https://github.com/mockito/mockito/wiki/FAQ